BranchBrowser.py

```diff
@@ -497,7 +500,121 @@ def update_repos(self, event):
+            if not repositories:
+                messagebox.showwarning("No Repositories", "No repositories found for the selected organization.")
+                return
+
+            repo_combobox['values'] = repositories
+            repo_combobox.set(self.default_repo) 
+
+        org_combobox.bind("<<ComboboxSelected>>", on_org_select)
+
+        repo_label = tk.Label(config_dialog, text="Select GitHub repository:")
+        repo_label.pack(pady=5)
+
+        repo_combobox = ttk.Combobox(config_dialog, values=[], width=30)
+        repo_combobox.set(self.default_repo)  
+        repo_combobox.pack(pady=5)
+
+        git_hostname_label = tk.Label(config_dialog, text="Enter GitHub hostname:")
+        git_hostname_label.pack(pady=5)
+
+        git_hostname_entry = tk.Entry(config_dialog, width=30)
+        git_hostname_entry.insert(0, 'github.com')  
+        git_hostname_entry.pack(pady=5)
+
+        def on_save():
+            new_org = org_combobox.get()
+            new_repo = repo_combobox.get()
+            new_git_hostname = git_hostname_entry.get()
+
+            if new_org and new_repo and new_git_hostname:
+                config = {
+                    'default_organization': new_org,
+                    'default_repository': new_repo,
+                    'GIT_HOSTNAME': new_git_hostname
+                }
+
+                self.save_config(config)
+                messagebox.showinfo("Success", "Configuration updated and saved.")
+                self.update_main_display(new_org, new_repo)
+                config_dialog.destroy()
+            else:
+                messagebox.showwarning("Input Error", "All fields must be provided.")
+
+        save_button = tk.Button(config_dialog, text="Save", command=on_save)
+        save_button.pack(pady=10)
+
+        cancel_button = tk.Button(config_dialog, text="Cancel", command=config_dialog.destroy)
+        cancel_button.pack(pady=5)
+
+    # Load configuration settings from 'config.json', or use defaults if file is missing or invalid
+    @staticmethod
+    def load_config():
+        config_path = os.path.join(os.path.dirname(__file__), "config.json")
+        if not os.path.exists(config_path):
+            print(f"Config file {config_path} not found. Using default values.")
+            return None
+        
+        try:
+            with open(config_path, "r") as config_file:
+                config = json.load(config_file)
+                return config
+        except json.JSONDecodeError:
+            print(f"Error decoding JSON from config file {config_path}. Using default values.")
+            return None  
+          
+    # Saves the provided configuration data to a file     
+    def save_config(self, config):
+        try:
+            with open(self.config_path, "w") as config_file:
+                json.dump(config, config_file, indent=4)
+            print(f"Config saved to {self.config_path}")
+        except Exception as e:

```

Comment:
Ovde mozemo lepse da hendlujemo exception, uvek je bolja praksa, proveriti sta se metoda moze da vrati kao tip exceptiona, da imamo tacan podatak sta je problem, pa onda ovaj genericki na kraju 
Ostavio sam ti primer za json.dump, ali svakako proveri

try:
    with open('config.json', 'w') as config_file:
        json.dump(config, config_file, indent=4)
except TypeError as e:
    print(f"TypeError: {e} - The object is not serializable to JSON.")
except IOError as e:
    print(f"IOError: {e} - There was an issue with file .")
except ValueError as e:
    print(f"ValueError: {e} - The data is not valid for JSON serialization.")
except Exception as e:
    print(f"Error saving config: {e}")